lib/src/dart_dev_cli.dart

```diff
@@ -89,12 +91,6 @@ dev(List<String> args) async {
 }
 
 void registerTask(TaskCli cli, TaskConfig config) {
-  if (_cliTasks[cli.command] != null) {
-    throw new ArgumentError(
```

Comment:
Actually, due to the way that `LocalCli.discover` works, it's not possible to detect local tasks overriding other local tasks without overhauling how that works... any local tasks with the same name trump other ones.

What about something like this? https://github.com/Workiva/dart_dev/commit/6c029f3aaf6bb0f2c3ad36c279757e883b3979b0